design/MultipleQueues.md

```diff
@@ -0,0 +1,83 @@
+# Multiple Queues
+
+Multiple queues is important for:
+
+- Explicit submission onto less capable queues, such as pure copy queues.
+- Explicit distribution of submissions on machines with multiple queues of the same family.
+
+## QueueFamily
+
+There are multiple families of queues.
+Availible families of queues are surfaced via `sequence<QueueFamily> Adapter.queueFamilies`.
+Support for `graphics`, `compute`, and `copy` operations are surfaced by the respective attributes on the QueueFamily.
+Queue families with `graphics` or `compute` will always have `copy`.
+There may be queue families in the future like D3D12's video decode queues, which do not support copy operations.
+If there are any families that support graphics, there is at least one family that supports both compute and graphics.
+The most capable family is always first in the `Adapter.queueFamilies` list.
+(It will either be a graphics+compute, or at least compute)
+Exposed adapters must have at least a compute family.
+Users can also create `QueueFamily`s.
+
+## Queue Creation
+
+Queues are created at Device creation time, and are exposed via `sequence<Queue> Device.queues`.
+Each element in `sequence<QueueFamily> WebGPUDeviceDescriptor.queueRequests` results in a corresponding element in `Device.queues`.
+If an user-provided QueueFamily does not match a family in `Adapter.queueFamilies`, a more capable family is used.
+If no available family can satisfy an user-provided family, device creation fails.
+
+## Synchronization
+
+Queues can have fences inserted into them, and any queue can wait on that fence to be complete.
+
+Synchronizing access to resources across queues should be done by telling a queue to wait until a fence is complete, followed by submitting command buffers that depend on that fence.
+Resource data may have multiple concurrent readers, or exclusively one writer, at a time.
+Resources may have multiple concurrent readers and writers, so long as all subranges satisfy many-read/single-write exclusion. [1]
+If different commands would violate this exclusion, the implmentation injects synchronization.
+If a command is submitted that will never be able to synchronize for exclusion without subsequent user commands, that Submit is refused.
+
```

Comment:
I think I know what you're trying to get at here.  But, in case, I am misunderstanding, please provide an example that would get rejected by the API.